Queued Messages and Interjections

hassoncs opened the pull request:
Introduces the ability to queue messages when the agent is busy and automatically send them when the agent becomes idle. This enhances the user experience by allowing continuous input without waiting for the agent to finish.

**Queuing** (queued send)
![2025-07-23 19 15 56](https://github.com/user-attachments/assets/f7ad55d8-e8ea-4d2d-9f95-682c335fb045)

**Interjection** (cancel, then send)
![2025-07-23 21 12 59](https://github.com/user-attachments/assets/2cf24787-21ec-467f-aeb9-53dbd9d799f9)


**Dev Notes**
	•	Implementation Choice: Decided to implement the feature on the client side using a React hook for easier management and to avoid conflicts with upstream changes from Roo

chrarnoldus commented:
I'm trying this out, but while the interjection works, queueing does not (I cannot submit anything when the AI is busy). Any idea what's going on?

hassoncs commented:
Thanks for taking a look at this @chrarnoldus - please hold off reviewing for now, I'm going to update it to work closer to Cursor 
![2025-07-08 17 24 30](https://github.com/user-attachments/assets/61025dd3-ed42-4b80-9653-98ffa702ac23)

Commit:
build(storybook): update decorator imports to use @storybook/react-vite

Commit:
build(deps): add framer-motion and update storybook dependencies

Commit:
feat(chat): Implement message queuing and interjection

This commit introduces message queuing and interjection functionality to the chat interface.
- **Message Queuing**: If the agent is busy, new messages entered by the user will now be queued and automatically sent when the agent becomes idle.
- **Interjection**: Pressing `Alt+Enter` (or `Option+Enter` on Mac) will now cancel the current task and immediately send the typed message, effectively interjecting the ongoing conversation. The queue will be paused after an interjection, requiring manual resumption.

This feature significantly improves the user experience by allowing users to continue typing and submitting messages even when the agent is busy, and provides a way to quickly interrupt and redirect the agent's focus.

New E2E tests have been added to cover both message queuing and interjection scenarios.
A new Storybook example and related components (`QueuedMessageList`, `QueuedMessageRow`, `useQueuedMessages` hook) have been added to facilitate development and testing of the queued message UI.

Commit:
'chore: WIP'

Commit:
feat(chat): Implement interjection and improve queued messages

Introduces `addToQueueAtFront` to `useQueuedMessages` hook, allowing messages to be immediately processed. This enables an interjection feature where users can send a new message while the agent is streaming, effectively canceling the current operation and sending the new message as the next in queue.

The `QueuedMessageList` component is updated to use a more robust animation for collapsing/expanding, replacing `Collapsible` with `motion.div` for better control over height transitions. Test cases for `useQueuedMessages` are expanded to cover the new `addToQueueAtFront` functionality and interjection scenarios.

Commit:
test(chat): Adjust queued message timing and update UI

Refactor `useQueuedMessages` to use a 500ms cooldown for sending messages, preventing double sends and aligning with test expectations. Update `QueuedMessageList` and `QueuedMessageRow` styling and component usage for better visual consistency and functionality.

The `useQueuedMessages` tests were updated to reflect the new 500ms cooldown, ensuring the queue behavior is correctly tested. The UI changes include:
- Adjusting the negative margin in `ChatTextArea` to use a CSS variable for better spacing control.
- Updating `QueuedMessageList` to use `useLayoutEffect` for height calculation, ensuring correct scroll behavior, and refining its border and background styles.
- Replacing manual button styling with `IconButton` component in `QueuedMessageList` and `QueuedMessageRow` for consistency.

Commit:
feat(chat): Refactor message queuing props and interjection logic

Refactor `ChatTextArea` props for message queuing to improve readability and consistency.
Adjust `handleInterjection` logic in `ChatView` to ensure interjection always cancels the current task, even if no new input is provided, and to correctly handle adding new input to the queue.
Update `onSend` call in `ChatTextArea` to ensure `resetHistoryNavigation` is called before `onSend`.

Commit:
test(e2e): Remove redundant message queue tests

Removed `message-queue-interjection.test.ts` and `message-queue-multiple-messages.test.ts` as their functionality is now covered by more comprehensive E2E tests. This streamlines the test suite and avoids duplication.

The `ChatView.tsx` change is a minor reordering of an existing `clearQueue()` call, which is a refactor and not directly related to the test removal.

Commit:
refactor(chat): Streamline message queuing logic and examples

Extract `createQueuedMessage` to centralize message creation.
Refactor `QueuedMessageListExample` to use a new `addSampleMessage` helper, reducing redundancy and improving readability.
Adjust `useQueuedMessages` to use the new `createQueuedMessage` for consistency.
Remove unnecessary console log and improve timeout cleanup in `useQueuedMessages`.

Commit:
refactor(chat): Clean up kilocode change markers in chat components

Commit:
refactor(ui): Remove unused variant prop from IconButton

The `variant` prop was removed from `IconButton` as it was only being used to apply a "subtle" style, which is now the default style for the component. This simplifies the component's API and reduces unnecessary complexity.

hassoncs commented on webview-ui/src/components/ui/IconButton.tsx:
```diff
@@ -0,0 +1,30 @@
+}
+
+export function IconButton({ size = "md", className, children, ...props }: IconButtonProps) {
+	return (
```
I just noticed we have a similar IconButton, we could combine them

hassoncs commented:
Closing this PR since I need to change the approach. It's too easy to lose messages with this current strategy. I need to update it to be event based. Stay tuned!
